GPR183.py

```diff
@@ -0,0 +1,184 @@
+    - length_bound (tuple, int, float): tuple of required range of sequences length (inclusive),
+    num or float if only higher border of the range is needed (exclusive).
+    - quality_threshold (int): int of lowest level of Q-score (inclusive).
+
+    Output:
+    - list of BioSeq records. Write file to .fastq
+    """
+    # Chech if PATH for input file is given
+    if input_path is None:
+        raise ValueError("You didn't enter any PATH to file")
+    # Chech if folder exist and create outout_path if not given
+    input_folder = input_path.rsplit('/', 1)[0]
+    input_name = input_path.rsplit('/', 1)[1]
+    is_exist = os.path.exists(f'{input_folder}/fastq_filtrator_resuls/')
+    if not is_exist:
+        os.makedirs(f'{input_folder}/fastq_filtrator_resuls/')
+    if output_filename is None:
+        output_path = f'{input_folder}/fastq_filtrator_resuls/{input_name}'
+    else:
+        output_path = f'{input_folder}/fastq_filtrator_resuls/{output_filename}.fastq'
+    # Create dict from FASTQ
+    seqs = list(SeqIO.parse(input_path, "fastq"))
+    # Check that this dict is not empty
+    if len(seqs) <= 0:
+        raise ValueError('There are no fastq sequences')
+    # Check if all given argumets have relevant type
+    gc_bound_type = isinstance(gc_bound, (tuple, int, float))
+    length_bound_type = isinstance(length_bound, (tuple, int, float))
+    quality_thr_type = isinstance(quality_threshold, (int, float))
+    if not (gc_bound_type and length_bound_type and quality_thr_type):
+        raise ValueError('Your arguments are not suitable!')
+    # Create filtered list
+    filtered_fastq = []
+
+    for line in seqs:
+        if isinstance(gc_bound, (int, float)):
+            gc_check = gc_fraction(line)*100 >= gc_bound
+        else:
+            gc_check = gc_fraction(line)*100 < gc_bound[0] or gc_fraction(line)*100 > gc_bound[1]
+        if isinstance(length_bound, (int, float)):
+            len_check = len(line) >= length_bound
+        else:
+            len_check = len(line) < length_bound[0] or len(line) > length_bound[1]
+        quality_check = sum(line.letter_annotations['phred_quality'])/len(line.letter_annotations['phred_quality']) < quality_threshold
+        if not (gc_check or len_check or quality_check):
+            filtered_fastq.append(line)
+
+    # Write  filtered data into new .fastq file
+    SeqIO.write(filtered_fastq, output_path, 'fastq')
+
+# Custom error
+class WrongSequence(ValueError):
+    pass
+
+
+class BiologicalSequence(str):
+    def __init__(self, sequence):
+        self.sequence = sequence
+
+    def __len__(self):
+        return len(self.sequence)
+    
+    def slice(self, start_index, stop_index):
+        return self.sequence[start_index:stop_index]
+    
+    def alphabet_checking(self):
+        if not set(self.sequence) <= set(type(self).dictionary.keys()):
+            raise WrongSequence('Wrong sequence')
+        return True
```

Comment:
Проверка - супер разумная, но, во-первых, она должна быть определена в следующем поколении классов. 
А во-вторых, она дублирует возврат True. Само условие уже подразумевает возвращение True или False, поэтому лучше бы сделать так 
```suggestion
    def alphabet_checking(self):
        return set(self.sequence) <= set(type(self).dictionary.keys())
```